tests/util/bft.py

```diff
@@ -359,11 +359,15 @@ def force_quorum_including_replica(self, replica_id, primary=0):
                            return
 
     async def is_state_transfer_complete(self, up_to_date_node, stale_node):
-            # Get the lastExecutedSeqNumber from a reference node
-            key = ['replica', 'Gauges', 'lastExecutedSeqNum']
-            last_exec_seq_num = await self.metrics.get(up_to_date_node, *key)
-            n = await self.metrics.get(stale_node, *key)
-            return n == last_exec_seq_num
+            try:
+                with trio.fail_after(seconds=5):
```

Comment:
I don't think the timeout needs to be that large, because the intention is that we are just checking when the system is up. It should be more like 1s or less. Larger timeouts are usually used with loops when waiting for something to happen. We are actually catching this error, presumably because we may loop using the value. A long timeout will make the tests take too long.

Reply:
Good point, thanks! 
-> resolved

Reply:
Hi Andrew, I tried applying your comment, but it seems to make the tests unstable. Reverting to my first proposal, trading-off test speed vs stability.

Let me know your thoughts.

Reply:
I'm very curious as to why this is is the case. If this is true, then we should be using a wait_for function in those scenarios, not a check that is supposed to be immediate. The goal of this function was always to be able to do an immediate check. When it's used the system should be up and responding with metrics. I actually think we should make the change in the C++ code such that we don't start the replica until the metrics server is up and running with valid metrics.

Reply:
How about making the BftTestNetwork.start_replica() async and waiting for the metrics to be available before we declare a replica started?

Reply:
The replica starting already doesn't guarantee anything, so changing the C++ code doesn't make any sense. We already start the replica server after the UDP server, but the metrics may not be published yet. So I was wrong about that fix.

However, the metrics not being available are why we have wait_for functions in our tests. If we need to wait for certain metrics we should. This is specific to the test though. We don't know what we need to wait for generically, and therefore don't want to wait for any specific metrics for all tests. Especially since they are published by different modules and potentially at different rates. 

My suggestion is that if we are seeing this problem, we should rename the function to wait_for or use the already existing wait_for_state_transfer_to_stop instead.

Reply:
Thanks Andrew, I'll give it another go tomorrow.

Reply:
Change of strategy today :) I got rid of is_state_transfer_complete() in favor of wait_for_state_transfer_to_stop(). After a chain of refactorings the two methods had become redundant.

Fingers crossed :)